token_dispenser_client/token_dispenser_client.py

```diff
@@ -0,0 +1,177 @@
+from botocore.exceptions import ClientError
+import boto3
+lambda_client = boto3.client('lambda')
+ssm = boto3.client('ssm')
+# ssm parameter store default path for TDS ARN
+_DEFAULT_SSM_PATH_: str = '/service/token-dispenser'
+logger = logging.getLogger(__name__)
+
+
+def get_parameter_by_name(name: str) -> str:
+    """
+
+    Args:
+        name: the ssm name point to TDS arn
+
+    Returns: tds arn as string
+    """
+    try:
+        if not name.endswith('/'):  # Check if it's a specific parameter name
+            response = ssm.get_parameter(
+                Name=name
+            )
+        return response['Parameter']['Value']
+    except KeyError as key_error:
+        logger.exception("Missing Parameter or Value key with name:%s response:%s",
+                         name, response)
+        raise key_error
+    except Exception as ex:  # Catch any other error.
+        logger.exception("An unexpected error occurred during get_parameter: "
+                         "name:%s response:%s", name, response)
+        raise ex
+
+
+def invoke_lambda(input_params_json, lambda_arn):
+    """
+    Invokes an AWS Lambda function with input parameters from a JSON string.
+
+    Args:
+        input_params_json (str): A JSON string containing input parameters.
+        lambda_arn (str): The ARN of the Lambda function to invoke.
+
+    Returns:
+        str: The JSON-formatted response from the Lambda function, or an error message.
+    """
+    if not lambda_arn:
+        return "Error: lambda_arn not provided"
+
+    try:
+        json.loads(input_params_json)
+    except json.JSONDecodeError:
+        return f'Error: Invalid JSON : {input_params_json}'
+
+    try:
+        response = lambda_client.invoke(
+            FunctionName=lambda_arn,
+            InvocationType='RequestResponse',  # Synchronous invocation
+            Payload=input_params_json
+        )
+
+        payload = response['Payload'].read().decode('utf-8')
+
+        # Check for function error
+        if 'FunctionError' in response:
+            raise RuntimeError(f"Error: Lambda function error: {payload}")
+
+        return payload
+    except ClientError as client_err:
+        logger.exception("Lambda invokation error occured %s",client_err)
+        raise client_err
+    except Exception as general_err:  # pylint: disable=broad-exception-caught
+        logger.exception("An unexpected general error occurred "
+                         "while invoking TDS: %s", general_err)
+        raise general_err
+
+
+def validate_input(client_id: str, minimum_alive_secs: int) -> List[str]:
+    """ validate the user inputs """
+    err_msgs = []
+    if client_id is None or client_id.strip() == '':
+        err_msgs.append('Error: client_id is required')
+
+    pattern = re.compile(r'^[a-zA-Z0-9]{3,32}$')
+    if client_id and not pattern.match(client_id):
+        err_msgs.append('client_id must be between length 3-32 '
+                        'with pattern [a-zA-Z0-9]{3,32}')
```

Comment:
Let's keep this on the service-side instead of the client-side. Easier to update later if it's only in one place